### ISSUE TITLE
Accessing CorLibTypeFactory (sometimes) causes the corlib module to "un-import" itself?

### ISSUE THREAD

**Issue by zsr2531** [Washi1337/AsmResolver#263]
## Describe the bug
Unfortunately, I had a hard time tracking this bug down, so I cannot give an exact description of it. Looking at it from the outside, using a type from `module.CorLibTypeFactory` will cause a `MemberNotImportedException` when you build the module.

## To Reproduce
The smallest file I could repro this on happened to be `dnSpy.dll` from dnSpyEx v6.1.9. Sorry for that, it's quite a big boi 😅 

```csharp
using AsmResolver.DotNet;
using AsmResolver.PE.DotNet.Metadata.Tables.Rows;

var module = ModuleDefinition.FromFile(@"C:\dnSpyEx\bin\dnSpy.dll");
var corlib = module.CorLibTypeFactory;

var type = new TypeDefinition(null, "Test", TypeAttributes.Class)
{
    BaseType = corlib.Object.ToTypeDefOrRef()
};

module.TopLevelTypes.Add(type);

// MemberNotImportedException occurs here
module.Write("out.dll");
```

## Expected behavior
No exception gets thrown and the module gets written to disk.

## Platform
 - OS: Tested on Win10, but probably all platforms are affected
 - AsmResolver Version: 4.8.0

**Comment by Washi1337** [comment]
Potential fix is pushed in #264. Could you maybe verify whether this solves your issue?

**Comment by zsr2531** [comment]
Yeah, the exception(s) are gone now. Thanks.

**Comment by Washi1337** [comment]
Perfect, thanks again for spotting. The fix will be included in 4.9.0.
